### DISCUSSION TITLE
Similar framework: reactor

### DISCUSSION THREAD

**Discussion by jbjuin** [tetra-framework/tetra#5] [Show and tell]
Hi Sam,
I just discovered this projet from the HN post today. I have been exploring this kind of work (reactive django components) since few months now and I use django reactor: https://github.com/edelvalle/reactor (version 2.2 though, not the last one) on a big project.

Tetra looks quite similar in usage but there are a few differences:

- js integration looks really clean in tetra,
- inlined templates/css/js shoudl be tested before saying anything about it :)
- blocks ! That is a great feature !

I will definitely take some time to read your code and try to see if we can port some of the features of Tetra to reactor too.
In reactor we built an app store (similar to vuex) and a router (similar to vue-router). They are not published yet but will be at some point.

I have a few questions: 

- any plan to integrate websockets communications ?
- do jinja works here ? If not do you plan to use it ? Using reactor we realized it was absolutely necessary for snappy reactions on big templates.
- is there an easy way for siblings components to communicate ?

Anyway it's great to see all those new "frameworks" extending django to ease the frontend creation.
Congrats !

**Comment by samwillis** [comment]
Hi!

Always good to hear from other people tackling the same problems. I haven’t seen your project, it looks impressive, I will certainly study it.

On your questions:

> any plan to integrate websockets communications ?

Yes, it an important feature. They will probably be a different component type that is “live”, so rather than the state being frozen and passed back and forth to the browser they will remain “live” on the server. No timeline yet, but would hope to have them in place for v1 this summer.

> do jinja works here ? If not do you plan to use it ? Using reactor we realized it was absolutely necessary for snappy reactions on big templates.

Not yet, but I would like to. Because of how Tetra hooks into the Django templating system for blocks, and keeps track of where they are defined, that will have to be replicated for Jinja. I suspect this won’t be in V1 but is certainly something that’s on the to do list for the future.

> is there an easy way for siblings components to communicate ?

Tetra subscribes to the idea of using the DOM as a message bus. Just as you do in Alpine or Vue.js. So it’s parse state down to child components, and have events bubble back up the DOM via Alpine event handlers. For two sibling components to communicate it would be via their parent in the browser.
